libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/internal/extensions/MapboxCarMapSurfaceExtensions.kt

```diff
@@ -0,0 +1,37 @@
+@file:JvmName("MapboxCarMapSurfaceEx")
+@file:OptIn(MapboxExperimental::class)
+
+package com.mapbox.androidauto.car.internal.extensions
+
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.Style
+import com.mapbox.maps.extension.androidauto.MapboxCarMapSurface
+import com.mapbox.maps.plugin.delegates.listeners.OnStyleLoadedListener
+
+internal fun MapboxCarMapSurface.handleStyleOnAttached(
+    block: (Style) -> Unit
+): OnStyleLoadedListener {
+    getStyle(block)
+
+    return OnStyleLoadedListener {
+        getStyle(block)
+    }.also {
+        mapSurface.getMapboxMap().addOnStyleLoadedListener(it)
+    }
+}
+
+internal fun MapboxCarMapSurface.handleStyleOnDetached(
+    listener: OnStyleLoadedListener?,
+    block: (Style) -> Unit
+) {
+    listener?.let { mapSurface.getMapboxMap().removeOnStyleLoadedListener(it) }
+    getStyle(block)
+}
+
+internal fun MapboxCarMapSurface.getStyle(block: (Style) -> Unit) {
```

Comment:
You should return `Style?` instead of accepting a lambda. This way the difference between `getStyle` and `getStyleAsync` is much clearer. 
The same with `handleStyleOnDetached`. With lambda it is not clear whether the block is invoked synchronously or not.